Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -328,9 +328,9 @@ namespace AZ::DocumentPropertyEditor
                 }
             }
 
-            ExtractLabel(attributes);
             if (access.GetType() == azrtti_typeid<AZStd::string>())
             {
+                ExtractLabel(attributes);
```

Comment:
I'm questioning now if this does anything. It might have been a coincidence that it was called before the str ptr case. Will double check before I make the change, and if anyone knows I'm wrong then please tell me!

Reply:
I'd err on the side of extract label being the general case, I think all property editor types can have a label.

Reply:
Commit incoming. Moving ExtractLabel back out to the general case.

Reply:
Actually, this PR is already treating ExtractLabel as the general case. It just extracted the code from ExtractLabel (calling it at the top of the `else` case and adds the label to the builder without modification if this isn't a container. If we move ExtractLabel out of the str ptr case, then we're back where we started.

If you think calling ExtractLabel in the general case needs to be more explicit then I can try to add a third case, but my intuition tells me that we actually have more than 3 cases here and things could snowball.